tests/pipelines/test_pipelines_common.py

```diff
@@ -875,84 +875,93 @@ def test_cfg(self):
 
     def test_callback_inputs(self):
         sig = inspect.signature(self.pipeline_class.__call__)
+        has_callback_tensor_inputs = "callback_on_step_end_tensor_inputs" in sig.parameters
+        has_callback_step_end = "callback_on_step_end" in sig.parameters
 
-        if not ("callback_on_step_end_tensor_inputs" in sig.parameters and "callback_on_step_end" in sig.parameters):
+        if not (has_callback_tensor_inputs and has_callback_step_end):
             return
 
         components = self.get_dummy_components()
         pipe = self.pipeline_class(**components)
         pipe = pipe.to(torch_device)
         pipe.set_progress_bar_config(disable=None)
-
         self.assertTrue(
             hasattr(pipe, "_callback_tensor_inputs"),
             f" {self.pipeline_class} should have `_callback_tensor_inputs` that defines a list of tensor variables its callback function can use as inputs",
         )
 
-        def callback_inputs_test(pipe, i, t, callback_kwargs):
-            missing_callback_inputs = set()
-            for v in pipe._callback_tensor_inputs:
-                if v not in callback_kwargs:
-                    missing_callback_inputs.add(v)
-            self.assertTrue(
-                len(missing_callback_inputs) == 0, f"Missing callback tensor inputs: {missing_callback_inputs}"
-            )
-            last_i = pipe.num_timesteps - 1
-            if i == last_i:
-                callback_kwargs["latents"] = torch.zeros_like(callback_kwargs["latents"])
+        def callback_inputs_success(pipe, i, t, callback_kwargs):
+            # interate over callback args
+            for tensor_name, tensor_value in callback_kwargs.items():
+                # check that we're only passing in allowed tensor inputs
+                assert tensor_name in pipe._callback_tensor_inputs
```

Comment:
this is nice to add! 

but it is also important that we testing what's missing - make sure all the tensors in listed in the pipeline class's `_callback_tensor_inputs` actual exist and we are able to fetch them for the callback function

maybe keep both?

Reply:
I can make that change 👍🏽